pkg/storage/pod.go

```diff
@@ -100,16 +100,9 @@ func (s *podStorage) GetMetrics(pods ...*metav1.PartialObjectMetadata) ([]metric
 }
 
 func (s *podStorage) Store(newPods *MetricsBatch) {
-	lastPods := make(map[apitypes.NamespacedName]PodMetricsPoint, len(newPods.Pods))
-	prevPods := make(map[apitypes.NamespacedName]PodMetricsPoint, len(newPods.Pods))
 	var containerCount int
 	for podRef, newPod := range newPods.Pods {
 		podRef := apitypes.NamespacedName{Name: podRef.Name, Namespace: podRef.Namespace}
-		if _, found := lastPods[podRef]; found {
-			klog.ErrorS(nil, "Got duplicate pod point", "pod", klog.KRef(podRef.Namespace, podRef.Name))
```

Comment:
This kind of error scenario will not be detected anymore and that might become an issue since we might be overwritting a pod entry without knowing